Sources/BinaryParsing/Parsers/FloatingPoint.swift

```diff
@@ -0,0 +1,212 @@
+    self = Self(bitPattern: bitPattern)
+  }
+
+  /// Creates a `Float` by parsing a value with the specified endianness from
+  /// the start of the given parser span.
+  ///
+  /// - Parameters:
+  ///   - input: The `ParserSpan` to parse from. If parsing succeeds, the start
+  ///     position of `input` is moved forward by 4 bytes.
+  ///   - endianness: The endianness to use when interpreting the parsed value.
+  /// - Throws: A `ParsingError` if `input` does not have enough bytes to store
+  ///   a `Float`.
+  public init(parsing input: inout ParserSpan, endianness: Endianness)
+    throws(ParsingError)
+  {
+    let bitPattern = try UInt32(parsing: &input, endianness: endianness)
+    self = Self(bitPattern: bitPattern)
+  }
+}
+
+extension Double {
+  /// Creates a `Double` by parsing a big-endian value from the start of the
+  /// given parser span.
+  ///
+  /// - Parameter input: The `ParserSpan` to parse from. If parsing succeeds,
+  ///   the start position of `input` is moved forward by 8 bytes.
+  /// - Throws: A `ParsingError` if `input` does not have enough bytes to store
+  ///   a `Double`.
+  public init(parsingBigEndian input: inout ParserSpan) throws(ParsingError) {
+    let bitPattern = try UInt64(parsingBigEndian: &input)
+    self = Self(bitPattern: bitPattern)
+  }
+
+  /// Creates a `Double` by parsing a little-endian value from the start of the
+  /// given parser span.
+  ///
+  /// - Parameter input: The `ParserSpan` to parse from. If parsing succeeds,
+  ///   the start position of `input` is moved forward by 8 bytes.
+  /// - Throws: A `ParsingError` if `input` does not have enough bytes to store
+  ///   a `Double`.
+  public init(parsingLittleEndian input: inout ParserSpan) throws(ParsingError)
+  {
+    let bitPattern = try UInt64(parsingLittleEndian: &input)
+    self = Self(bitPattern: bitPattern)
+  }
+
+  /// Creates a `Double` by parsing a value with the specified endianness from
+  /// the start of the given parser span.
+  ///
+  /// - Parameters:
+  ///   - input: The `ParserSpan` to parse from. If parsing succeeds, the start
+  ///     position of `input` is moved forward by 8 bytes.
+  ///   - endianness: The endianness to use when interpreting the parsed value.
+  /// - Throws: A `ParsingError` if `input` does not have enough bytes to store
+  ///   a `Double`.
+  public init(parsing input: inout ParserSpan, endianness: Endianness)
+    throws(ParsingError)
+  {
+    let bitPattern = try UInt64(parsing: &input, endianness: endianness)
+    self = Self(bitPattern: bitPattern)
+  }
+}
+
+#if !(os(Windows) || os(Android) || ($Embedded && !os(Linux) && !(os(macOS) || os(iOS) || os(watchOS) || os(tvOS)))) && (arch(i386) || arch(x86_64))
+
+// Local development on a non-supported platform is super fun...
+//
+//#else
+//public struct Float80: Sendable, Equatable {
+//  init(sign: FloatingPointSign, exponentBitPattern: UInt, significandBitPattern: UInt64) {}
+//}
```

Comment:
As with `Float16`, if anything what I would provide here is a fallback bailable parser that tries to parse the `Float80` into a `Double` (what to do for out-of-range values TBD).